src/components/Api.js

```diff
@@ -0,0 +1,80 @@
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+  checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "GET",
+      headers: this._headers,
+    }).then(this.checkResponse);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this.checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Thanks for the advice, if its okay I would like to just keep it how I have it just because your way slightly confuses me but good to know I can do it like that for future related stuff.

Reply:
> Thanks for the advice, if its okay I would like to just keep it how I have it just because your way slightly confuses me but good to know I can do it like that for future related stuff.

This is how you can use it in your code

```js

  _request(url, options) {
    return fetch(url, options).then(this.checkResponse)
  }


  getInitialCards() {
    return this._request(`${this._baseUrl}/cards`, {
      method: "GET",
      headers: this._headers,
    })
  }

  getProfileInfo() {
    return this._request(`${this._baseUrl}/users/me`, {
      method: "GET",
      headers: this._headers,
    })
  }
```

Reply:
Ahh okay, that makes a little more sense. Okay I will add that change, thank you!